packages/plugin-core/src/views/common/preview/PreviewPanel.ts

```diff
@@ -52,8 +66,9 @@ export class PreviewPanel implements PreviewProxy, vscode.Disposable {
     @inject("wsRoot") private wsRoot: URI,
     private wsUtils: WSUtilsWeb,
     private webViewUtils: WebViewUtils,
-    @inject("IPreviewPanelConfig") private config: IPreviewPanelConfig,
-    @inject("INoteRenderer") private noteRenderer: INoteRenderer
+    @inject("vaults") private vaults: DVault[],
+    @inject("DendronConfig") private dendronConfig: DendronConfig,
+    @inject("ReducedDEngine") private engine: ReducedDEngine
   ) {
     this._linkHandler = linkHandler;
```

Comment:
nit: ts-trick I learned recently - you can use the syntax below to avoid needing to set these properties in the constructor body. 

```typescript
    @inject("IPreviewLinkHandler") private _linkHandler: IPreviewLinkHandler,
    @inject("ITextDocumentService") private _textDocumentService: ITextDocumentService,
```

This will then cause some es-lint rules to complain about empty constructor, but I've disabled them in a pending PR.